internal/states/instance_object_src.go

```diff
@@ -58,6 +58,11 @@ type ResourceInstanceObjectSrc struct {
 	// state, or to save as sensitive paths when saving state
 	AttrSensitivePaths []cty.PathValueMarks
 
+	// TransientPathValueMarks helps propagate non-sensitive marks through the
+	// internal representation of a state, without being serialized into the final
+	// state file.
+	TransientPathValueMarks []cty.PathValueMarks
```

Comment:
The code in instance_object.go does not filter out sensitive marks.  I don't think that's a problem functionally, but it's made a bit confusing by the comment.

Reply:
That's indeed confusing, thanks for catching!

[dd8e250](https://github.com/opentofu/opentofu/pull/2679/commits/dd8e2509e3a3f88750b4ba86d4b68e4266bf842e)

Reply:
Thanks for fixing it. I was already having some comments to add related to `TransientPathValueMarks`.